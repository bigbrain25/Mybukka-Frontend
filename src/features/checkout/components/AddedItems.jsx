import React, { useState, useEffect } from 'react';

import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Price from 'Components/badge/Price';
import Button from 'Components/button/Button';
import itemDetails from '../InputAttribute/inputData.json';

import './addedItem.scss';

const Edit = ({ handleClick }) => (
  <Button
    text="EDIT"
    type="button"
    handleClick={handleClick}
    classNames="cart-action-button-primary"
  />
);

const Remove = ({ handleClick }) => (
  <Button
    text="REMOVE"
    type="button"
    handleClick={handleClick}
    classNames="cart-action-button-default"
  />
);

const OrderTray = ({ handleRemove, handleEdit, name, price }) => (
  <div className="d-flex order-tray">
    <div className="col p-0">
      <h5 className="item-name">1 Ã— {name}</h5>
      <div className="action-item-section">
        <Edit handleClick={handleEdit} />
        <Remove handleClick={handleRemove} />
      </div>
    </div>
    <div className="p-0">
      <Price price={price} classNames="price-badge-black" />
    </div>
  </div>
);

const AddedItem = () => {
  const [suggestedItemsInTray, setTray] = useState(itemDetails.suggestedTray);

  const removeItemHandler = (index) => {
    // get the state
    const trayItem = [...suggestedItemsInTray];
    trayItem.splice(index, 1);
    // set the suggestedItemsInTray state
    setTray(trayItem);
    // replace in json but no need if redux state
    itemDetails.suggestedTray = trayItem;
  };

  useEffect(() => {
    setTray(itemDetails.suggestedTray);
  });

  return (
    <div className="cart-menu">
      <div className="cart-bukka-details">
        <h5 className="cart-bukka-name">{"Mel's Drive-In"}</h5>
        <h5 className="cart-bukka-view-menu">
          <Link className="text-success view-menu-text" to="/feed">
            VIEW MENU
          </Link>
        </h5>
      </div>

      {suggestedItemsInTray.map((item, index) => (
        <OrderTray
          name={item.name}
          price={item.price}
          key={item.name}
          handleRemove={() => removeItemHandler(index)}
        />
      ))}
    </div>
  );
};

export default AddedItem;

OrderTray.propTypes = {
  handleRemove: PropTypes.func.isRequired,
  handleEdit: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired
};

Remove.propTypes = {
  handleClick: PropTypes.func.isRequired
};

Edit.propTypes = {
  handleClick: PropTypes.func.isRequired
};
