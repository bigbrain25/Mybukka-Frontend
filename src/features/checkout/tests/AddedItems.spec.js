import React from 'react';

import AddedItems
  from '../components/AddedItems';

describe('Area to explore component', () => {
  const props = {
    handleClick: jest.fn()
  };
  const wrapper = mount(<AddedItems {...props} />);

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
