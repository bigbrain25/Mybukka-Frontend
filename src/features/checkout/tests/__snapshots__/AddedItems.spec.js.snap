// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Area to explore component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="cart-menu"
      >
        <div
          class="cart-bukka-details"
        >
          <h5
            class="cart-bukka-name"
          />
          <h5
            class="cart-bukka-view-menu"
          >
            <a
              class="text-success view-menu-text"
              href="/bukka/undefined"
            >
              VIEW MENU
            </a>
          </h5>
        </div>
        <div
          class="d-flex order-tray"
        >
          <div
            class="col p-0"
          >
            <h5
              class="item-name"
            >
              1 × 
              name
            </h5>
            <div
              class="action-item-section"
            >
              <button
                class="cart-action-button-primary"
                data-dismiss=""
                data-target=""
                data-testid="button"
                data-toggle=""
                id="u8PdR4bbp"
                type="button"
              >
                EDIT
              </button>
              <button
                class="cart-action-button-default"
                data-dismiss=""
                data-target=""
                data-testid="button"
                data-toggle=""
                id="u8PdR4bbp"
                type="button"
              >
                REMOVE
              </button>
            </div>
          </div>
          <div
            class="p-0"
          >
            <h4
              class="price-badge-black"
            >
              ₦
              90
              .00
            </h4>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="cart-menu"
    >
      <div
        class="cart-bukka-details"
      >
        <h5
          class="cart-bukka-name"
        />
        <h5
          class="cart-bukka-view-menu"
        >
          <a
            class="text-success view-menu-text"
            href="/bukka/undefined"
          >
            VIEW MENU
          </a>
        </h5>
      </div>
      <div
        class="d-flex order-tray"
      >
        <div
          class="col p-0"
        >
          <h5
            class="item-name"
          >
            1 × 
            name
          </h5>
          <div
            class="action-item-section"
          >
            <button
              class="cart-action-button-primary"
              data-dismiss=""
              data-target=""
              data-testid="button"
              data-toggle=""
              id="u8PdR4bbp"
              type="button"
            >
              EDIT
            </button>
            <button
              class="cart-action-button-default"
              data-dismiss=""
              data-target=""
              data-testid="button"
              data-toggle=""
              id="u8PdR4bbp"
              type="button"
            >
              REMOVE
            </button>
          </div>
        </div>
        <div
          class="p-0"
        >
          <h4
            class="price-badge-black"
          >
            ₦
            90
            .00
          </h4>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
